tracer/src/Datadog.Trace.Tools.Runner/RunHelper.cs

```diff
@@ -36,6 +38,11 @@ public static int Execute(ApplicationContext applicationContext, CommandContext
                 }
             }
 
+            var arguments = args.Count > 1 ? string.Join(' ', args.Skip(1).ToArray()) : null;
+
+            // Fix: wrap arguments containing spaces with double quotes ( "[arg with spaces]" )
+            FixDoubleQuotes(ref arguments);
```

Comment:
Really 🤔 Shouldn't customers be doing that. This doesn't seem like something we should be doing, as we don't know for sure where the spaces should be? 🤔 It seems like you're assuming there's only one argument (your unit tests are all with one argument too)

Reply:
This also seems to me like something the user would be responsible for, so we don't make incorrect assumptions

Reply:
The reason for this is because .NET removes all the quotes from the arguments string, even if the customer adds them, and in a CLI scenario than can cause problems (due to invalid arguments).

### Example:

![image](https://user-images.githubusercontent.com/69803/169557657-a5e7393f-bea7-461e-9a8d-ec7cf77b5bcf.png)

#### Without quotes:

![image](https://user-images.githubusercontent.com/69803/169557802-621ea1a3-50f7-4453-9013-8ff00d51b13e.png)

#### After the method:

![image](https://user-images.githubusercontent.com/69803/169557916-993e7310-8d27-46be-894a-49033da9f9ab.png)

I realized this problem when trying to apply the normal code coverage in linux + datadog code coverage at the same time.


I wrote a test suite for this method here: https://github.com/DataDog/dd-trace-dotnet/blob/91bb5b295475eb741fa263883e4093f7be5fcb7c/tracer/test/Datadog.Trace.Tools.Runner.Tests/RunHelperTests.cs

Reply:
If you know a way to extract the command line in a raw format, It will fix this behavior and I won't need any of this handling.

Reply:
> The reason for this is because .NET removes all the quotes from the arguments string, even if the customer adds them

What, that's madness! 🤦‍♂️ 

I still disagree that we should _need_ to be doing this, as it seems _so_ hard to get right, but I guess we don't have a choice 😬 

Given that, IMO we need a _lot_ more unit tests around it. AFAICT we currently are only really testing 2 different patterns `--arg:*` and `-a:*`. We should try it with multiple arguments, some of which need escaping, some of which don't, space instead of colon, args which contain `'` and/or `"` etc

Reply:
I agree, I will create another PR with more test cases.